opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -53,59 +47,47 @@ def collect(self) -> Optional[_PointVarT]:
         pass
 
 
-class SynchronousSumAggregation(
-    _InstrumentMonotonicityAwareAggregation, Aggregation[Sum]
-):
-    def __init__(self, instrument_is_monotonic: bool):
-        super().__init__(instrument_is_monotonic)
+class SumAggregation(Aggregation[Sum]):
+    def __init__(
+        self,
+        instrument_is_monotonic: bool,
+        instrument_temporality: AggregationTemporality,
+    ):
+        super().__init__()
+        self._instrument_is_monotonic = instrument_is_monotonic
+
         self._value = 0
+
         self._start_time_unix_nano = _time_ns()
+        self._instrument_temporality = instrument_temporality
 
     def aggregate(self, measurement: Measurement) -> None:
+
         with self._lock:
             self._value = self._value + measurement.value
 
-    def collect(self) -> Optional[Sum]:
+    def collect(self) -> Sum:
         """
         Atomically return a point for the current value of the metric and
         reset the aggregation value.
         """
-        now = _time_ns()
+        if self._instrument_temporality is AggregationTemporality.DELTA:
+            now = _time_ns()
 
-        with self._lock:
-            value = self._value
-            start_time_unix_nano = self._start_time_unix_nano
+            with self._lock:
+                value = self._value
+                start_time_unix_nano = self._start_time_unix_nano
 
-            self._value = 0
-            self._start_time_unix_nano = now + 1
-
-        return Sum(
-            aggregation_temporality=AggregationTemporality.DELTA,
-            is_monotonic=self._instrument_is_monotonic,
-            start_time_unix_nano=start_time_unix_nano,
-            time_unix_nano=now,
-            value=value,
-        )
+                self._value = 0
+                self._start_time_unix_nano = now + 1
 
-
-class AsynchronousSumAggregation(
-    _InstrumentMonotonicityAwareAggregation, Aggregation[Sum]
-):
-    def __init__(self, instrument_is_monotonic: bool):
-        super().__init__(instrument_is_monotonic)
-        self._value = None
-        self._start_time_unix_nano = _time_ns()
-
-    def aggregate(self, measurement: Measurement) -> None:
-        with self._lock:
-            self._value = measurement.value
-
-    def collect(self) -> Optional[Sum]:
-        """
-        Atomically return a point for the current value of the metric.
-        """
-        if self._value is None:
-            return None
+            return Sum(
```

Comment:
We need to reset `self._value` in this case as well.

Reply:
Hmm I see there is a bit of an edge case here. If this aggregation doesn't receive and measurements, it needs to return the value from the last collection interval. OTOH, if it does receive measurements, it shouldn't add them to the previous windows.

I think this was another reason for having the return type be `Optional`, to handle this case. Any thought's on how best to handle this?

Reply:
> We need to reset `self._value` in this case as well.

I think we [are](https://github.com/open-telemetry/opentelemetry-python/pull/2379/files/fefb92205c259d94127196bf1c5bf16ae2af8b06#diff-753a117844eaabe8d1982225dd1537b48afa4ad164f6f5752a4a64fa35a3a450R81).

Reply:
> If this aggregation doesn't receive and measurements, it needs to return the value from the last collection interval.

If the aggregation `aggregate` method is not called at all and then `collect` is, it will return a `0`-valued `Sum` object. Do we consider `0` to be the value from the last collection interval in this situation? I can imagine that not being representative or this situation (`aggregate` not being at all then `collect` being called) because it will have the same value as `aggregate` being called with a `0`-valued measurement. We could instead return a `None`-valued `Sum` object instead to indicate this (Which saves us from having to use `Optional` in the return type).

> OTOH, if it does receive measurements, it shouldn't add them to the previous windows.

Don't understand, what do you mean with "previous windows"?

Reply:
> I think we [are](https://github.com/open-telemetry/opentelemetry-python/pull/2379/files/fefb92205c259d94127196bf1c5bf16ae2af8b06#diff-753a117844eaabe8d1982225dd1537b48afa4ad164f6f5752a4a64fa35a3a450R81).

Link not working :( 

> I think we [are](https://github.com/open-telemetry/opentelemetry-python/pull/2379/files/fefb92205c259d94127196bf1c5bf16ae2af8b06#diff-753a117844eaabe8d1982225dd1537b48afa4ad164f6f5752a4a64fa35a3a450R81).

> Do we consider 0 to be the value from the last collection interval in this situation?

For Synchronous Counter, this is fine and makes sense. For Asynchronous counters, it doesn't make sense and it should never go down i.e. we should continue reporting the previous value if the callback didn't yield any measurements this time around.

> We could instead return a None-valued Sum object instead to indicate this (Which saves us from having to use Optional in the return type).

👍 that could work

> Don't understand, what do you mean with "previous windows"?

Meaning the value from the previous collection.

Reply:
> Link not working :(

Sorry, I mean [this](https://github.com/open-telemetry/opentelemetry-python/pull/2379/files#diff-753a117844eaabe8d1982225dd1537b48afa4ad164f6f5752a4a64fa35a3a450R81):

``` python
    def collect(self) -> Sum:
￼        """
￼        Atomically return a point for the current value of the metric and
￼        reset the aggregation value.
￼        """
        if self._instrument_temporality is AggregationTemporality.DELTA:
￼            now = _time_ns()
￼
￼            with self._lock:
￼                value = self._value
￼                start_time_unix_nano = self._start_time_unix_nano
￼
￼                self._value = 0  # <- this line
￼                self._start_time_unix_nano = now + 1
￼
```
 
> For Synchronous Counter, this is fine and makes sense. For Asynchronous counters, it doesn't make sense and it should never go down i.e. we should continue reporting the previous value if the callback didn't yield any measurements this time around. 

I mean a situation where we have an instrument (Asynchronous or Synchronous) and its corresponding aggregation `collect` method is called before the first call to `aggregate`. But now that I think more about this, that won't happen, right? Before that `collect` method gets called by the `MetricReaderStorage` the `callback` will be called, right? Like it happens [here](https://github.com/open-telemetry/opentelemetry-python/pull/2301/files#diff-693118fd4a59966d0e9d2499bb9704ce0af972e3390e712b3a7488e0848009bbR122).

Reply:
@aabmass I introduced the changes requested in the SIG, I hope they are right, please take a look :v: